sormas-backend/src/test/java/de/symeda/sormas/backend/performance/PerformanceLogAnalysisGenerator.java

```diff
@@ -0,0 +1,331 @@
+/*
+ * SORMAS® - Surveillance Outbreak Response Management & Analysis System
+ * Copyright © 2016-2022 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package de.symeda.sormas.backend.performance;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Stack;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.io.FilenameUtils;
+import org.jetbrains.annotations.NotNull;
+
+public class PerformanceLogAnalysisGenerator {
+
+	private static final Pattern startPattern =
+		Pattern.compile("([^ ]* [^ ]*) TRACE *(.*) d.s.s.b.u.PerformanceLoggingInterceptor - Started: ([^ ]*) with parameters .*$");
+	private static final Pattern finishPattern =
+		Pattern.compile("([^ ]* [^ ]*) DEBUG *(.*) d.s.s.b.u.PerformanceLoggingInterceptor - Finished in ([0-9]*) ms: (.*)$");
+
+	//@formatter:off
+    private static final String HTML_HEADER = "<html>\n" +
+            "<header>\n" +
+            "<title>Performance Log Analysis</title>" +
+            "<style>\n" +
+            "* {font-family: Verdana, Arial, Helvetica, sans-serif;}\n" +
+            "a {font-weight: bold; color: #666; text-decoration: none;}\n" +
+            "a:hover {color: #ff9900;}\n" +
+            "th {font-weight: bold;}\n" +
+            "th,td {padding: 0 5px 0 5px; vertical-align: top;}\n" +
+            "tr:nth-child(odd) {background: #EEE}\n" +
+            "tr:nth-child(even) {background: #FFF}\n" +
+            "</style>\n" +
+            "</header>\n" +
+            "<body>\n" +
+            "<table>\n";
+    private static final String HTML_FOOTER = "</table>\n" +
+            "</body>\n" +
+            "</html>\n";
+    //@formatter:on
+
+	public static final int TIME_TRESHOLD_YELLOW = 100;
+	public static final int TIME_TRESHOLD_ORANGE = 300;
+	public static final int TIME_TRESHOLD_RED = 1000;
+
+	public static final String OUTPUT_DIRECTORY = "target/performance/";
+
+	private Map<String, Stack<String>> callstacks;
+	private Map<String, MethodStats> methodStats;
```

Comment:
Why instance variable? Better move to method if not needed outside

Reply:
Obsolete and removed.